include/net/socket.h

```diff
@@ -312,7 +317,7 @@ static inline int getaddrinfo(const char *host, const char *service,
 
 static inline void freeaddrinfo(struct zsock_addrinfo *ai)
```

Comment:
Nothing to complain about the change itselft, but I just wonder - shouldn't we use aliased types with aliased functions? I mean `struct addrinfo` instead of `struct zsock_addrinfo`?
I see some inconsistency in here - for instance `select` mixes aliased `timeval` with non-aliased `zsock_fd_set` in the same call.

Reply:
> but I just wonder - shouldn't we use aliased types with aliased functions?

Well, maybe we should. I wouldn't consider that technically important matter, because we alias types with #define, so it's all the same. For beautification purposes might be good idea to have a pass there.

But this patch doesn't add or changes anything like that, so please approve.